clients/spark/core/src/main/scala/io/treeverse/clients/GarbageCollector.scala

```diff
@@ -434,6 +440,39 @@ object GarbageCollector {
     spark.close()
   }
 
+  private def logRunId(
+      runID: String,
+      storageType: String,
+      storageNamespace: String,
+      hc: Configuration,
+      region: String
+  ): Unit = {
+    val uri = new URI(storageNamespace)
+    val runIDMarkerInputStream = new ByteArrayInputStream(new Array[Byte](0))
+    if (storageType == StorageUtils.StorageTypeS3) {
+      val bucket = uri.getHost
+      val s3Client = getS3Client(hc, bucket, region, StorageUtils.S3.S3NumRetries)
+      val meta = new ObjectMetadata()
+      meta.setContentLength(0)
+      s3Client.putObject(bucket,
+                         getRunIDMarkerLocation(runID, uri.getPath),
+                         runIDMarkerInputStream,
+                         meta
+                        )
+    } else if (storageType == StorageUtils.StorageTypeAzure) {
+      val storageAccountUrl = StorageUtils.AzureBlob.uriToStorageAccountUrl(uri)
+      val storageAccountName = StorageUtils.AzureBlob.uriToStorageAccountName(uri)
+      val containerName = StorageUtils.AzureBlob.uriToContainerName(uri)
+      val blobClient =
+        Azure.getBlobContainerClient(hc, storageAccountUrl, storageAccountName, containerName)
+      val pathArray = uri.getPath.split("/")
+      val key = pathArray.slice(2, pathArray.length).reduce((a1, a2) => a1 + "/" + a2)
+      blobClient
+        .getBlobClient(getRunIDMarkerLocation(runID, key))
+        .upload(runIDMarkerInputStream, 0)
+    }
```

Comment:
This should be an additional method on some polymorphic object.  Maybe even on BulkRemover?

Reply:
sounds good, but not as part of the process of the bulk remover (I don't want it to write the ID multiple times by each executed task)